src/scene_graph/scene_graph.cpp

```diff
@@ -197,31 +197,83 @@ namespace wr
 			for (unsigned int i = 0; i < m_mesh_nodes.size(); ++i) {
 
 				auto node = m_mesh_nodes[i];
-				auto it = m_batches.find(node->m_model);
```

Comment:
Please split this code  into functions where possible and add comments as the code is not immediate to understand